src/manifolds/Sphere.jl

```diff
@@ -290,8 +290,9 @@ log(::AbstractSphere, ::Any...)
 
 function log!(M::AbstractSphere, X, p, q)
     cosθ = real(dot(p, q))
+    cosθ = abs(cosθ) > 1 ? sign(cosθ) * one(cosθ) : cosθ
     if cosθ ≈ -1 # appr. opposing points, return deterministic choice from set-valued log
-        fill!(X, 0)
+        fill!(X, 0.0)
```

Comment:
This change should not be necessary, since 0 will be promoted as necessary.

Reply:
ah, ok, thanks, I wasn't sure (or confused in between).